tests/runner/src/main/java/org/mule/test/runner/api/AetherClassPathClassifier.java

```diff
@@ -637,12 +649,32 @@ private void buildPluginUrlClassification(Artifact artifactToClassify, ClassPath
     logger.debug("Artifacts {} identified a plugin dependencies for plugin {}", pluginArtifactDependencies, artifactToClassify);
     pluginArtifactDependencies.stream()
         .map(artifact -> {
-          String artifactClassifierLessId = toId(artifact);
-          if (!artifactsClassified.containsKey(artifactClassifierLessId)) {
+          Optional<ArtifactClassificationNode> artifactClassificationNodeOptional =
+              findArtifactClassified(artifactsClassified, artifact);
+          if (!artifactClassificationNodeOptional.isPresent()) {
             buildPluginUrlClassification(artifact, context, directDependenciesFilter, artifactsClassified,
                                          rootArtifactRemoteRepositories);
+          } else {
+            if (!areCompatibleVersions(artifactClassificationNodeOptional.get().getArtifact().getVersion(),
+                                       artifact.getVersion())) {
+              throw new IllegalStateException(
+                                              format("Incompatible version of artifacts found: %s and %s",
+                                                     toId(artifactClassificationNodeOptional.get().getArtifact()),
+                                                     toId(artifact)));
+            }
+            logger.debug("Checking for highest version of artifact, already discovered: '{}' versus: '{}'",
```

Comment:
WARN?

Reply:
Changed next logger sentence where actually the replacing is happening